python/marvin/contrib/vacs/galaxyzoo3d.py

```diff
@@ -0,0 +1,864 @@
+            the ellipse equal to the number of points belonging to that cluster.
+        spiral_mask (numpy.array):
+            Pixel mask (same shape as image) of the spiral arm location(s).  The value for the pixels is the number
+            of overlapping polygons at that location.
+        bar_mask (numpy.array):
+            Pixel mask (same shape as image) of the bar location.  The value for the pixels is the number of
+            overlapping polygons at that location.
+        metadata (astropy.Table):
+            Table containing metadata about the galaxy.
+        ifu_size (int):
+            Size of IFU
+        center_clusters (astropy.Table):
+            Position for identified galaxy center(s) in both image pixels and (RA, DEC)
+        num_centers (int):
+            Number of galaxy centers identified
+        star_clusters (astropy.Table):
+            Position for identified forground star(s) in both image pixels and (RA, DEC)
+        num_stars (int):
+            Number of forground stars identified
+        center_star_classifications (astropy.Table):
+            Raw GZ3D classifications for center(s) and star(s)
+        num_center_star_classifications (int):
+            Total number of classifications made for either center(s) or star(s)
+        num_center_star_classifications_non_blank (int):
+            Total number of non-blank classifications made for either center(s) or star(s)
+        spiral_classifications (astropy.Table):
+            Raw GZ3D classifications for spiral arms
+        num_spiral_classifications (int):
+            Total number of spiral arm classifications made
+        num_spiral_classifications_non_blank (int):
+            Total number of non-blank spiral arm classifications made
+        bar_classifications (astropy.Table):
+            Raw GZ3D classifications for bars
+        num_bar_classifications (int):
+            Total number of bar classifications made
+        num_bar_classifications_non_blank (int):
+            Total number of non-blank bar classifications made
+        cube (marvin.tools.cube.Cube):
+            Marvin Cube object
+        maps (marvin.tools.maps.Maps)
+            Marvin Maps object
+        center_mask_spaxel (numpy.array):
+            The center_mask projected into spaxel space
+        star_mask_spaxel (numpy.array):
+            The star_mask projected into spaxel space
+        spiral_mask_spaxel (numpy.array):
+            The spiral_mask projected into spaxel space
+        bar_mask_spaxel (numpy.array):
+            The bar_mask projected into spaxel space
+        other_mask_spaxel (numpy.array):
+            A mask for spaxel not contained in any of the other spaxel masks
+    '''
+    def __init__(self, filename, cube, maps):
+        # get the subject id from the filename
+        self.subject_id = filename.split('/')[-1].split('_')[-1].split('.')[0]
+        # read in the fits file
+        self.hdulist = fits.open(filename)
```

Comment:
I wonder if you make this a contextmanager, i.e. `with fits.open(filename) as self.hdulist` it will handle edge cases where something breaks in the code, at least the file will be closed.  Then you could remove the explicit `close` method.

Reply:
good idea, I have updated the code.